collections/distinct_by_test.ts

```diff
@@ -0,0 +1,126 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+import { assertEquals } from "https://deno.land/std@0.100.0/testing/asserts.ts";
+import { distinctBy } from "./distinct_by.ts";
+
+function distinctByTest<I>(
+  array: Array<I>,
+  selector: (element: I) => unknown,
+  expected: Array<I>,
+  message?: string,
+) {
+  const actual = distinctBy(array, selector);
+  assertEquals(actual, expected, message);
+}
+
+Deno.test({
+  name: "[collections/distinctBy] identities on empty array",
+  fn() {
+    distinctByTest(
+      [],
+      () => {},
+      [],
+    );
+  },
+});
+
+Deno.test({
+  name: "[collections/distinctBy] gets head on noop selector",
+  fn() {
+    distinctByTest(
+      [25, "asdf", true],
+      () => {},
+      [25],
+    );
+  },
+});
+
+Deno.test({
+  name:
+    "[collections/distinctBy] removes duplicates and preserves order on identity",
+  fn() {
+    distinctByTest(
+      [true, "asdf", 4, "asdf", true],
+      (it) => it,
+      [true, "asdf", 4],
+    );
+    distinctByTest(
+      [null, undefined, null, "foo", undefined],
+      (it) => it,
+      [null, undefined, "foo"],
+    );
+    distinctByTest(
```

Comment:
Yep 🙈 I probably copied the test three times and forgot to adapt the third - it was a long copy-paste evening^^

I cannot change it ln vacation. Feel free to delete the duplicat test or change it to a new one. I will donit when I return otherwise :-)